psi4/src/psi4/libmints/wavefunction.cc

```diff
@@ -235,6 +235,110 @@ void Wavefunction::deep_copy(const Wavefunction *other)
+    factory_ = std::shared_ptr<MatrixFactory>(new MatrixFactory);
+
+    Dimension c1_nsopi = Dimension(1);
+    c1_nsopi[0] = other->nsopi_.sum();
+    factory_->init_with(c1_nsopi, c1_nsopi);
+
+    // Need to re-generate AO2SO_ because the new SO basis is different
+    // than the old one (lower symmetry)
+    std::shared_ptr<PetiteList> pet(new PetiteList(basisset_, integral_));
+    AO2SO_ = pet->aotoso();
+
+    psio_ = other->psio_; // We dont actually copy psio
+    memory_ = other->memory_;
+    nalpha_ = other->nalpha_;
+    nbeta_ = other->nbeta_;
+    nfrzc_ = other->nfrzc_;
+
+    print_ = other->print_;
+    debug_ = other->debug_;
+    density_fitted_ = other->density_fitted_;
+
+    energy_ = other->energy_;
+    efzc_ = other->efzc_;
+    variables_ = other->variables_;
+
+    // collapse all the Dimension objects down to one element
+    doccpi_.init(1, other->doccpi_.name());
+    doccpi_[0] = other->doccpi_.sum();
+    soccpi_.init(1, other->soccpi_.name());
+    soccpi_[0] = other->soccpi_.sum();
+    frzcpi_.init(1, other->frzcpi_.name());
+    frzcpi_[0] = other->frzcpi_.sum();
+    frzvpi_.init(1, other->frzvpi_.name());
+    frzvpi_[0] = other->frzvpi_.sum();
+    nalphapi_.init(1, other->nalphapi_.name());
+    nalphapi_[0] = other->nalphapi_.sum();
+    nbetapi_.init(1, other->nbetapi_.name());
+    nbetapi_[0] = other->nbetapi_.sum();
+    nsopi_.init(1, other->nsopi_.name());
+    nsopi_[0] = other->nsopi_.sum();
+    nmopi_.init(1, other->nmopi_.name());
+    nmopi_[0] = other->nmopi_.sum();
+
+    nso_ = other->nso_;
+    nmo_ = other->nmo_;
+    nirrep_ = 1;
+
+    same_a_b_dens_ = other->same_a_b_dens_;
+    same_a_b_orbs_ = other->same_a_b_orbs_;
+
+
+    S_ = factory_->create_shared_matrix("S");
+    S_->remove_symmetry(other->S(), other->aotoso());
+
+    /// Below is not set in the typical constructor
+    H_ = factory_->create_shared_matrix("One-electron Hamiltonian");
+    H_->remove_symmetry(other->H(), other->aotoso());
+
+    if (other->Ca_) Ca_ = other->Ca_subset("AO", "ALL");
+    if (other->Cb_) Cb_ = other->Cb_subset("AO", "ALL");
+    if (other->Da_) Da_ = other->Da_subset("AO");
+    if (other->Db_) Db_ = other->Db_subset("AO");
+    if (other->Fa_) Fa_ = other->Fa_subset("AO");
+    if (other->Fb_) Fb_ = other->Fb_subset("AO");
+    if (other->epsilon_a_) epsilon_a_ =
+        other->epsilon_subset_helper(other->epsilon_a_, other->nsopi_, "AO", "ALL");
+    if (other->epsilon_b_) epsilon_b_ = 
+        other->epsilon_subset_helper(other->epsilon_b_, other->nsopi_, "AO", "ALL");
+
+
+    // these are simple SharedMatrices of size 3*natom_, etc., so should
+    // not depend on symmetry ... can just copy them
+    if (other->gradient_) gradient_ = other->gradient_->clone();
+    if (other->hessian_) hessian_ = other->hessian_->clone();
+    if (other->tpdm_gradient_contribution_)
+        tpdm_gradient_contribution_ = other->tpdm_gradient_contribution_->clone();
+
```

Comment:
This is more of a tag for me, the internal `array` and `variable` maps were not copied over until #736. Should probably add that here also. Although, what to do with the `array` objects that might be irreped is interesting.